apps/problem-sets/4-ui-customization/4.1-tabs/app-solution.py

```diff
@@ -1,66 +1,38 @@
-import pandas as pd
-from pathlib import Path
-from plots import temp_distirbution, daily_error
-
-infile = Path(__file__).parent / "weather.csv"
-weather = pd.read_csv(infile)
-weather["error"] = weather["observed_temp"] - weather["forecast_temp"]
-
-
-app_ui = ui.page_fluid(
-    ui.panel_title("Weather error"),
-    ui.layout_sidebar(
-        ui.panel_sidebar(
-            ui.input_date_range("dates", "Date", start="2022-01-01", end="2022-01-30"),
-            ui.input_selectize(
-                "cities",
-                "Select Cities",
-                weather["city"].unique().tolist(),
-                selected="BUFFALO",
-                multiple=True,
-            ),
-            ui.input_slider("alpha", "Plot Alpha", value=0.5, min=0, max=1),
-            width=3,
-        ),
-        ui.panel_main(
-            ui.navset_tab(
-                ui.nav(
-                    "Error",
-                    ui.output_plot("error_distribution"),
-                    ui.output_plot("error_by_day"),
-                ),
-                ui.nav("Data", ui.output_data_frame("data")),
-            )
-        ),
-    ),
-)
-
-
-def server(input, output, session):
-    @reactive.Calc
-    def filtered_data():
-        df = weather.copy()
-        df = df[df["city"].isin(input.cities())]
-        df["date"] = pd.to_datetime(df["date"])
-        dates = pd.to_datetime(input.dates())
-        df = df[(df["date"] > dates[0]) & (df["date"] <= dates[1])]
-        return df
-
-    @output
-    @render.plot
-    def error_distribution():
-        return temp_distirbution(filtered_data())
-
-    @output
-    @render.plot
-    def error_by_day():
-        return daily_error(filtered_data(), input.alpha())
-
-    @output
+from shiny.express import render, ui, input
+from data_import import df
+from plots import plot_auc_curve, plot_precision_recall_curve
+from shinywidgets import render_plotly
+
+with ui.sidebar():
+    ui.input_select(
+        "account",
+        "Account",
+        choices=[
+            "Berge & Berge",
+            "Fritsch & Fritsch",
+            "Hintz & Hintz",
+            "Mosciski and Sons",
+            "Wolff Ltd",
+        ],
+    )
+
+with ui.nav_panel("Training Dashboard"):
+    @render_plotly
+    def precision_recall_plot():
+        account_subset = df[df["account"] == input.account()]
+        return plot_precision_recall_curve(
+            account_subset, "is_electronics", "training_score"
+        )
+
+
+    @render_plotly
+    def auc_plot():
+        account_subset = df[df["account"] == input.account()]
+        return plot_auc_curve(account_subset, "is_electronics", "training_score")
+
+with ui.nav_panel("Data"):
     @render.data_frame
-    def data():
-        return filtered_data()
-
-
-app = App(app_ui, server)
+    def table():
+        account_subset = df[df["account"] == input.account()]
+        account_counts = account_subset.groupby("sub_account").size().reset_index(name="counts")
```

Comment:
I think it's better here to display a big data frame, because it makes the second tab a bit more compelling.